arvidn/libtorrent
# Enhance CMake build

1. Add tools to CMake build
1. Replace custom 'shared' option with the standard `BUILD_SHARED_LIBS`  flag
2. Replace `foreach()` calls for source items with `target_sources()` and the generator expression `$<JOIN:>`.
3. Remove build types definition: makes no sense for  single-configuration generators, and is populated automatically for multi-configuration ones.
4. Add feature summary, printed at the end of the configure stage. Example:
```
-- The following features have been enabled:

 * BUILD_SHARED_LIBS, build libtorrent as a shared library
 * build_tests, build tests
 * build_examples, build examples
 * build_tools, build tools
 * python-bindings, build python bindings
 * dht, enable support for Mainline DHT
 * encryption, Enables encryption in libtorrent
 * exceptions, build with exception support
 * extensions, Enables protocol extensions
 * i2p, build with I2P support
 * ipv6, build with IPv6 support
 * logging, build with logging

-- The following features have been disabled:

 * static_runtime, build libtorrent with static runtime

-- The following RECOMMENDED packages have been found:

 * Iconv, GNU encoding conversion library, <https://www.gnu.org/software/libiconv/>
   Convert strings between various encodings
 * OpenSSL, Full-strength general purpose cryptography library, <https://www.openssl.org/>
   Provides HTTPS support to libtorrent
```

5. Enahnce compiler flags management for static runtime by utilizing functions from the [ucm project](https://github.com/onqtam/ucm)
6. Copy almost all options from the Jamfile.
7. Use C++14 mode If compiler supports that.
8. Rename variables in FindIconv.cmake to match those in the same module  appeared in CMake 3.11.
9. Try to use config mode first while looking for the package in the example FindLibtorrentRasterbar.cmake.
10. Rename targets and config files for consistency:  CMake does check both suffixes anyway (-config and Config), but since the project name is "libtorrent-rasterbar", the "-config" suffix looks  better.
11. Install the FindLibtorrentRasterbar.cmake module.
